### PR TITLE
Add Rust solution for First Question

### PR DESCRIPTION
**Pull request by MohsenBg**
### This PR includes the following updates
* Implemented  first question solution in Rust using a greedy approach to determine if we can reach the last index.
* Added unit tests covering edge cases, large inputs, and failure scenarios.
* Removed the 'other player' directory from the branch MohsenBg as it's no longer needed.

### CODE REVIEW

**Review by Rezishon — APPROVED** [review]
Fine.

File: MohsenBg/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# MohsenBg Archive Questions 📚
+This is a collection of the questions I've solved. You can check how I solve them, and if you have any questions or ideas 💡, feel free to share them with me! 
+
+### List of Questions
+* [jump_game](./jump_game/README.md) 🎮
```
Comment (Rezishon) [inline MohsenBg/README.md:5]: Ok

File: MohsenBg/jump_game/src/lib.rs
Comment (Rezishon) [inline MohsenBg/jump_game/src/lib.rs:1]: Noce job!

File: MohsenBg/jump_game/README.md
Comment (Rezishon) [inline MohsenBg/jump_game/README.md:1]: Well done.

**Review by Rezishon — APPROVED** [review]
Well done. Hope you enjoyed it.

**Comment by Rezishon** [comment]
MohsenBg solved jump_game question
